pubsub/pubsub.go

```diff
@@ -565,85 +565,71 @@ type SubscriptionURLOpener interface {
 //
 // The zero value is a multiplexer with no registered schemes.
 type URLMux struct {
-	subscriptionSchemes map[string]SubscriptionURLOpener
-	topicSchemes        map[string]TopicURLOpener
+	subscriptionSchemes urlSchemeMap
```

Comment:
The "zero value" comment above is no longer true IIUC. It's a nice property. Maybe the name could be passed to each of (`Register`, `FromURL`, `FromString`) instead, so that the zero value of urlSchemeMap works?

Reply:
We should also add a test for the zero value.

Reply:
Passing the name to each method.

`TestURLMux` does create a zero value. Did you mean something else?

Reply:
I'm not sure how that test passed with your previous commit. It looks good now though.

Reply:
Ah, I see, it was just missing the name; it always handled `m` being `nil`.